### PR TITLE
NAS-150 measure usage

### PR DESCRIPTION
**Pull request by krejzd**
New sdk-backend-spi method getMeasureReferencingObjects added and used to get the data for measure editor

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5025/ : SUCCESS in 47s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6693/ : SUCCESS in 16m 54s

File: libs/sdk-backend-bear/src/backend/workspace/measures/index.ts [hunk 1]
```diff
@@ -154,4 +157,39 @@ export class BearWorkspaceMeasures implements IWorkspaceMeasuresService {
...
         return measure;
     }
+
+    async getMeasureReferencingObjects(ref: ObjRef): Promise<IMeasureReferencing> {
+        const uri = await objRefToUri(ref, this.workspace, this.authCall);
+        const objectId = getObjectIdFromUri(uri);
+
+        const measures = await this.authCall(async (sdk) => {
+            const usedBy = await sdk.xhr.getParsed<{ entries: GdcMetadata.IObjectXrefEntry[] }>(
+                `/gdc/md/${this.workspace}/usedby2/${objectId}?types=metric`,
```
Comment (kandl) [inline libs/sdk-backend-bear/src/backend/workspace/measures/index.ts:167]: Is it possible to use `sdk.md.getObjectUsedBy` method in bear client, or is there something missing/different? Also, you should be able to query visualizationObject & metric metadata objects at once like this. (by providing both types in options.types param). If the method is not suitable, please parallelise these 2 requests by Promise.all() (they don't depend on each other)

File: libs/sdk-backend-spi/src/workspace/measures/index.ts [hunk 2]
```diff
@@ -40,4 +44,12 @@ export interface IWorkspaceMeasuresService {
      * @returns promise of undefined
      */
     deleteMeasure(measureRef: ObjRef): Promise<void>;
+
+    /**
+     * Get all metadata objects which uses specified object(ie. object is used by these objects) by a given reference.
```
Comment (kandl) [inline libs/sdk-backend-spi/src/workspace/measures/index.ts:49]: Nitpick: missing space before the bracket

File: libs/sdk-backend-tiger/src/backend/workspace/measures/index.ts [hunk 3]
```diff
@@ -149,4 +153,42 @@ export class TigerWorkspaceMeasures implements IWorkspaceMeasuresService {
...
+                    workspaceId: this.workspace,
+                },
+                { query: filterReferencingObj as any }, // return only measures that have a link to the given id in their visualizationObjects
+            ).then(MetadataUtilities.mergeEntitiesResults),
+        );
+
+        const filterReferencingObjX = {
+            filter: `metric.id==${id}`, // RSQL format of querying data
+        };
+        const measures = await this.authCall((client) =>
```
Comment (kandl) [inline libs/sdk-backend-tiger/src/backend/workspace/measures/index.ts:177]: Could we parallelise these 2 requests (1 for the measures, 1 for the insight) by Promise.all() ?

File: libs/sdk-backend-tiger/src/convertors/fromBackend/MetricConverter.ts [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
 import { isInheritedObject } from "./utils";
 
-export function convertMetricFromBackend(data: JsonApiMetricOutDocument): IMeasureMetadataObject {
-    const { id, attributes } = data.data;
+export function convertMetricFromBackend(
+    data: JsonApiMetricOutDocument | JsonApiMetricOutWithLinks,
+): IMeasureMetadataObject {
+    let id: string;
+    let attributes: JsonApiMetricInAttributes;
+    if ("data" in data) {
```
Comment (kandl) [inline libs/sdk-backend-tiger/src/convertors/fromBackend/MetricConverter.ts:18]: What is purpose of this?

Comment (krejzd) [reply libs/sdk-backend-tiger/src/convertors/fromBackend/MetricConverter.ts:18]: The data I receive has another type, so i attempted to reuse the method to get MeasureMetadata object from it

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5029/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6698/ : SUCCESS in 19m 24s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5048/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6722/ : SUCCESS in 18m 30s

File: libs/sdk-backend-tiger/src/convertors/fromBackend/MetricConverter.ts (hunk 4 above)
Comment (kandl) [reply libs/sdk-backend-tiger/src/convertors/fromBackend/MetricConverter.ts:18]: Ok, so please let's not depend on "data" property, but let's create a type-guard for JsonApiMetricOutWithLinks type.

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1693

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1439/ : SUCCESS in 1m 09s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1716/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/501/ : SUCCESS in 12m 27s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6743/ : SUCCESS in 15m 57s

**Pull request merged**